test/integration_test.py

```diff
@@ -764,6 +767,30 @@ def _list_sources():
             self.assertEqual(_list_sources(), unauthenticated_sources)
         self.assertTrue(unauthenticated_sources <= all_sources)
 
+        managed_access_sources = {
+            source['sourceId']
+            for source in all_sources - unauthenticated_sources
+        }
+
+        if managed_access_sources:
```

Comment:
It looks like all of this should still pass even if `managed_access_sources == {}`. I don't have a strong opinion here, but I'd probably remove the conditional here.